src/error-boundary/internal.tsx

```diff
@@ -0,0 +1,125 @@
+    suppressed: boolean | typeof RootSuppressed;
+  }
+>({
+  onError: () => {},
+  suppressed: RootSuppressed,
+});
+
+interface InternalErrorBoundaryProps
+  extends SomeRequired<ErrorBoundaryProps, 'suppressNested'>,
+    InternalBaseComponentProps {}
+
+export function InternalErrorBoundary({
+  children,
+  suppressNested,
+  __internalRootRef,
+  ...props
+}: InternalErrorBoundaryProps) {
+  const context = useContext(ErrorBoundariesContext);
+  const thisSuppressed = context.suppressed === true;
+  const nextSuppressed = suppressNested || thisSuppressed;
+
+  const [forcedError, setForcedError] = useState(false);
+  const setElement = (node: null | HTMLDivElement) => {
+    if (node) {
+      if (!node.__awsui__) {
+        node.__awsui__ = {};
+      }
+      node.__awsui__.forceError = () => setForcedError(true);
+      node.__awsui__.clearForcedError = () => setForcedError(false);
+    }
+  };
+  const mergedRef = useMergeRefs(setElement, __internalRootRef);
+
+  return (
+    <div ref={mergedRef} className={styles['error-boundary']} data-awsui-boundary-id={props.errorBoundaryId}>
+      {!thisSuppressed ? (
+        <ErrorBoundaryImpl {...props} forcedError={forcedError}>
+          <ErrorBoundariesContext.Provider value={{ ...props, suppressed: nextSuppressed }}>
+            {children}
+          </ErrorBoundariesContext.Provider>
+        </ErrorBoundaryImpl>
+      ) : (
+        <>{children}</>
+      )}
+    </div>
+  );
+}
+
+interface BuiltInErrorBoundaryProps {
+  children: React.ReactNode;
+  wrapper?: (content: React.ReactNode) => React.ReactNode;
+  suppressNested?: boolean;
+}
+
+export function BuiltInErrorBoundary({ wrapper, suppressNested = false, children }: BuiltInErrorBoundaryProps) {
+  const context = useContext(ErrorBoundariesContext);
+  const thisSuppressed = context.suppressed === true || context.suppressed === RootSuppressed;
+  const nextSuppressed = suppressNested || thisSuppressed;
+  return !thisSuppressed ? (
+    <ErrorBoundaryImpl {...context} wrapper={wrapper}>
+      <ErrorBoundariesContext.Provider value={{ ...context, suppressed: nextSuppressed }}>
+        {children}
+      </ErrorBoundariesContext.Provider>
+    </ErrorBoundaryImpl>
+  ) : (
+    <>{children}</>
+  );
+}
+
+interface ErrorBoundaryState {
+  hasError: boolean;
+}
+
+interface ErrorBoundaryImplProps extends ErrorBoundaryProps {
+  wrapper?: (content: React.ReactNode) => React.ReactNode;
+  forcedError?: boolean;
+}
+
+class ErrorBoundaryImpl extends Component<ErrorBoundaryImplProps, ErrorBoundaryState> {
+  constructor(props: ErrorBoundaryImplProps) {
+    super(props);
+    this.state = { hasError: false };
+  }
+
+  static getDerivedStateFromError() {
+    return { hasError: true };
+  }
+
+  componentDidCatch(error: Error, errorInfo: React.ErrorInfo): void {
+    this.props.onError?.({ error, errorInfo, errorBoundaryId: this.props.errorBoundaryId });
```

Comment:
Since the `onError` prop is mandatory, is the `?` necessary?

Reply:
Great catch - I will update that.